tests/ArgentPonyWarcraftClient.Integration.Tests/TestUtilities/RequestResultExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Linq;
+using System.Text.Json;
+using FluentAssertions.Json;
+using Newtonsoft.Json.Linq;
+
+namespace ArgentPonyWarcraftClient.Integration.Tests
+{
+    public static class RequestResultExtensions
+    {
+        /// <summary>
+        /// Asserts that the <paramref name="requestResult"/> matches the
+        /// specified <paramref name="expectedJsonContent"/>.
+        /// </summary>
+        /// <typeparam name="T">The model type.</typeparam>
+        /// <param name="requestResult">The request result.</param>
+        /// <param name="expectedJsonContent">The expected JSON content.</param>
+        public static void ShouldMatchJsonContent<T>(this RequestResult<T> requestResult, string expectedJsonContent)
+        {
+            // Compare values as JObjects, removing null values on both actual and result. This
+            // addresses an issues where objects like spell tooltips sometimes use a null value and sometimes
+            // exclude a property, e.g. { "cast_time": "Passive" } vs. { "cast_time": "Instant", "power_cost": null }
+            var actualJsonValue = JObject.Parse(JsonSerializer.Serialize(requestResult.Value)).RemoveNullProperties();
+            var expectedJsonValue = JObject.Parse(expectedJsonContent).RemoveNullProperties();
+
+            actualJsonValue.Should().BeEquivalentTo(expectedJsonValue);
+        }
```

Comment:
This is very nice.  🥇

Reply:
Thanks! Honestly I'm super glad I added this because I genuinely missed some data on the `Covenant` class on the first pass 😅  TDD FTW